packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -470,11 +493,29 @@ function useRestoreFocus(scopeRef: RefObject<Element[]>, restoreFocus: boolean,
         document.removeEventListener('keydown', onKeyDown, true);
       }
 
-      if (restoreFocus && nodeToRestore && isElementInScope(document.activeElement, scopeRef.current)) {
+      // if we already lost focus to the body and this was the active scope, then we should attempt to restore
+      if (
+        restoreFocus
+        && nodeToRestore
+        && (
+          isElementInScope(document.activeElement, scopeRef.current)
+          || (document.activeElement === document.body && activeScope === scopeRef)
+        )
+      ) {
+        // freeze the focusScopeTree so it persists after the raf, otherwise during unmount nodes are removed from it
+        let clonedTree = focusScopeTree.clone();
```

Comment:
An alternative to cloning the whole tree would be to only clone the path from `nodeToRestore` to the root here, since that's all you care about.

Reply:
true, but i can't imagine there will be that many focus scopes on the screen that this would ever be a performance issue.
still, happy to change it